src/scalability/data-sharding.md

```diff
@@ -0,0 +1,23 @@
+# Data sharding
+
+## What is the partition
+
+For fault tolerance, TiKV replicates data to multiple nodes via Raft Consensus. For large datasets or high query throughput, having all the data is managed by one Raft state machine is not efficient and flexible, so we need to break the data up into partitions, also known as *sharding*, in TiKV a partition is called a `Region`.
+
+With partitioning, we can move partitions to balance the load among all nodes of a cluster. How to decide which partitions to store on which nodes isn't covered here, it will be discussed later in a chapter related to PD.
+
+## Partitioning
+
+Different partitions can be placed on different nodes, thus a large dataset can be distributed across many disks, and the query load can be distributed across many processors. But partitioning suffers from the issue of *hot spots*, where high load may end up on one partition but leave the other nodes idle. To avoid that, we can simply assign records to nodes randomly, but when reading a particular item there is no fast way of knowing which node is on. 
```

Comment:
```suggestion
Different partitions can be placed on different nodes, and thus a large dataset can be distributed across many disks, and the query load can be distributed across many processors. But partitioning suffers from the issue of *hot spots*, where high load may end up on one partition but leave the other nodes idle. To avoid that, we can simply assign records to nodes randomly, but when reading a particular item there is no fast way of knowing which node is on. 
```

Reply:
Note: "thus" is an adverb rather than a conjunction.